### DISCUSSION TITLE
Use Rectangular selection to extract text

### DISCUSSION THREAD

**Discussion by matar3** [RyotaUshio/obsidian-pdf-plus#198] [Q&A]
I know it is already possible to quote text by highlighting and copying it, whereas the Rectangular Selection is used for the function "Copy embed link to rectangular selection", that makes a kind of snapshot of the selection letting you paste in an obsidian .md file. 
I think it would be great if it were possible to use Rectangular Selection to catch the text in the selection in same way as highlighting does: the difference would be that drawing a rectangle is much easier and less boring that manually put the mouse cursor on a line of text and then manually dragging the selection following the structure of the lines. I guess such a thing would  imply a 2 step process, that should be done by the pdf++ program: first identify the lines inside the rectangular selection, and then extract text from them.

**Comment by RyotaUshio** [comment]
To be honest, I don't get your point. You don't need to put your cursor at the precise beginning position of the target text to start selection, right?

https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/3c14ec0f-471a-4f6c-af51-ff47e069dde4

Taking this into consideration, text selection by the rectangle tool will result in more steps required to accomplish the same goal with less degree of freedom. Am I missing something?

**Reply by matar3** [reply]
Try with two-column texts... However, i tried myself and u're right, it's already possible to use classic highlighting selection in a fashion similar to rectangular selection.

**Reply by matar3** [reply]
I don't want to insist, but please try yourself that what you showed in your video just works to select the first lines of a page, whereas it doesn't work (even in the lorem ipsum pdf) even for first lines of different paragraphs in same page. 
I'm attaching you a video of text selection feature in foxit, that is the kind of thing i make reference to (also consider that the standard tool in foxit is the hand, that if dragged in white space just scrolls the page, whereas if dragged on the text it selects the text).
P.S. I'm sorry but in this recording you can't see the mouse pointer; however, i wanted to show that in the beginning there's the hand, that scrolls the page and then selects text, and in the end there is the "Text selection tool" that is the rectangular selection i like.
https://github.com/RyotaUshio/obsidian-pdf-plus/assets/31188460/81ed3782-bda8-4105-97dd-3edebf514523
